judge/contest_format/ecoo.py

```diff
@@ -0,0 +1,121 @@
+            return
+
+        if not isinstance(config, dict):
+            raise ValidationError('ECOO-styled contest expects no config or dict as config')
+
+        for key, value in config.items():
+            if key not in cls.config_defaults:
+                raise ValidationError('unknown config key "%s"' % key)
+            if not isinstance(value, type(cls.config_defaults[key])):
+                raise ValidationError('invalid type for config key "%s"' % key)
+            if not cls.config_validators[key](value):
+                raise ValidationError('invalid value "%s" for config key "%s"' % (value, key))
+
+    def __init__(self, contest, config):
+        self.config = self.config_defaults.copy()
+        self.config.update(config or {})
+        self.contest = contest
+
+    def update_participation(self, participation):
+        cumtime = 0
+        points = 0
+        format_data = {}
+
+        with connection.cursor() as cursor:
+            cursor.execute('''
+            SELECT (
+                SELECT MAX(ccs.points)
+                    FROM judge_contestsubmission ccs LEFT OUTER JOIN
+                         judge_submission csub ON (csub.id = ccs.submission_id)
+                    WHERE ccs.problem_id = cp.id AND ccs.participation_id = %s AND csub.date = MAX(sub.date)
+            ) AS `score`, MAX(sub.date) AS `time`, cp.id AS `prob`, (
+                SELECT COUNT(ccs.id)
+                    FROM judge_contestsubmission ccs LEFT OUTER JOIN
+                         judge_submission csub ON (csub.id = ccs.submission_id)
+                    WHERE ccs.problem_id = cp.id AND ccs.participation_id = %s AND csub.result NOT IN ('IE', 'CE')
+            ) AS `subs`, cp.points AS `max_score`
+                FROM judge_contestproblem cp INNER JOIN
+                     judge_contestsubmission cs ON (cs.problem_id = cp.id AND cs.participation_id = %s) LEFT OUTER JOIN
+                     judge_submission sub ON (sub.id = cs.submission_id)
+                GROUP BY cp.id
+            ''', (participation.id, participation.id, participation.id))
+
+            for score, time, prob, subs, max_score in cursor.fetchall():
+                time = from_database_time(time)
+                dt = (time - participation.start).total_seconds()
+                if self.config['cumtime']:
+                    cumtime += dt
+
+                bonus = 0
+                if score > 0:
+                    # First AC bonus
+                    if subs == 1 and score == max_score:
+                        bonus += self.config['first_ac_bonus']
+                    # Time bonus
+                    if self.config['time_bonus']:
+                        bonus += (participation.end_time - time).total_seconds() // 60 // self.config['time_bonus']
+                    points += bonus
+
+                format_data[str(prob)] = {'time': dt, 'points': score, 'bonus': bonus}
+                points += score
+
+        participation.cumtime = 0
```

Comment:
If you are going to keep track of cumtime for optional time breaking, this should be `cumtime` and not `0`.